File: Core/Entities/Product.cs
namespace Core.Entities
{
    public class Product : BaseEntity
    {
        public string Name { get; set; }
        public decimal Price { get; set; }
        public string Description { get; set; }
        public bool Available { get; set; } = true;
        public List<ProductPhoto> ProductPhotos { get; set; }
        public ProductItemClass ProductItemClass { get; set; }
        public UserProduct UserProduct { get; set; }
    }
}

File: client/src/app/shared/models/product.ts
export interface Product {
  id: number
  name: string
  price: number
  description: string
  stockQuantity: number
  productPhotos: ProductPhoto[]
  productItemClass: ProductItemClass
  userProduct: UserProduct
}

interface ProductPhoto {
  photoId: number
  photo: Photo
}

interface Photo {
  id: number
  url: string
}

interface ProductItemClass {
  productId: number
  itemClassId: number
  itemClass: ItemClass
}

interface ItemClass {
  id: number
  name: string
}

interface UserProduct {
  userId: number
  user: User
  productId: number
}

interface User {
  id: number
  phoneNumber: string
  email: string
  firstName: string
  lastName: string
  dateOfBirth: string
  facebook: string
  instagram: string
}
